backend/daemon/api/documents/v1alpha/documents.go

```diff
@@ -947,6 +947,85 @@ func (api *Server) ListAccountPublications(ctx context.Context, in *documents.Li
 	return out, nil
 }
 
+// MergeChanges implements the corresponding gRPC method. It merges changes and publishes them.
+func (api *Server) MergeChanges(ctx context.Context, in *documents.MergeChangesRequest) (*documents.Document, error) {
+	me, err := api.getMe()
+	if err != nil {
+		return nil, err
+	}
+
+	if in.TargetDocumentId == "" {
+		return nil, fmt.Errorf("Target ID is mandatory")
+	}
+	eid := hyper.EntityID(in.TargetDocumentId)
+
+	source, err := hyper.Version(in.SourceDocumentVersion).Parse()
+	if err != nil {
+		return nil, err
+	}
+
+	doc, err := api.CreateDraft(ctx, &documents.CreateDraftRequest{
+		ExistingDocumentId: in.TargetDocumentId,
+		Version:            in.TargetDocumentVersion,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if doc.Id != eid.String() {
+		return nil, fmt.Errorf("Internal error. Draft entity and doc don't match")
+	}
+	entity, err := api.blobs.LoadEntity(ctx, eid)
+	if err != nil {
+		return nil, err
+	}
+	ch, err := api.blobs.GetDraft(ctx, eid)
+	if err != nil {
+		return nil, err
+	}
+
+OUTER:
+	for _, dep := range source {
+		for _, alreadyDep := range ch.Deps {
+			if dep == alreadyDep {
+				continue OUTER
+			}
+		}
+		ch.Deps = append(ch.Deps, dep)
+	}
+
+	del, err := api.getDelegation(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	hb, err := entity.CreateChange(entity.NextTimestamp(), me.DeviceKey(), del, map[string]any{
+		// TODO(juligasa): Remove this and add in terra.go NewChange if deps>1 and no patch don't error since its a merge
+		"isDraft": true,
+	}, hyper.WithDeps(ch.Deps), hyper.WithAction(hyper.ActionUpdate))
+
+	if err != nil {
+		return nil, err
+	}
+	/*
+		_, err = api.UpdateDraft(ctx, &documents.UpdateDraftRequest{
+			DocumentId: eid.String(),
+			Changes:    hb.Decoded,
+		})
+		if err != nil {
+			return nil, err
+		}*/
+	if err := api.blobs.SaveDraftBlob(ctx, eid, hb); err != nil {
```

Comment:
This fails since there already exists a draft for that document. But if I try to update the draft instead (Commented code above) I don't know how to insert the newly created blob as a change.

Reply:
I don't understand. What are you trying to do here? If you want to make a draft out of the merged version, but there's already a draft, it can't be done. If there's a draft then the frontend should rebase the draft to incorporate outstanding changes, I believe.

Or what's the desired behavior here? @horacioh @juligasa @ericvicenti